### ISSUE TITLE
"No guest options available for arch 'x86_64'" with test driver

### ISSUE THREAD

**Issue by clnperez** [kimchi-project/kimchi#591]
Reverting 640b257175d119c71f756ba08cdae599310e26e8 takes care of this issue:

$ sudo ./run_tests.sh test_mockmodel.MockModelTests.test_vm_info
WARNING: no 'numpy' module, HyBi protocol will be slower
Loading YumUpdate features.
Loaded plugins: fastestmirror, ibm-repository, langpacks, refresh-packagekit
WebSocket server settings:
  - Listen on :64667
  - Flash security policy server
  - Web server. Web root: /home/christy/git/kimchi-upstream/ui/pages/websockify
  - SSL/TLS support
  - Deny non-SSL/TLS connections
  - proxying from :64667 to targets in /var/lib/kimchi/vnc-tokens

[06/Feb/2015:10:06:28] ENGINE Bus STARTING
[06/Feb/2015:10:06:28] ENGINE Started monitor thread 'Autoreloader'.
[06/Feb/2015:10:06:28] ENGINE Serving on 127.0.0.1:51529
*** Running feature tests ***
127.0.0.1 - - [06/Feb/2015:10:06:28] "HEAD /images/icon-fedora.png HTTP/1.1" 200 4449 "" ""
127.0.0.1 - - [06/Feb/2015:10:06:28] "GET /images/icon-fedora.png HTTP/1.1" 206 4449 "" ""
127.0.0.1 - - [06/Feb/2015:10:06:28] "HEAD /images/icon-fedora.png HTTP/1.1" 200 4449 "" ""
libvirt: Test Driver error : Domain not found
libvirt: Domain Config error : internal error: No guest options available for arch 'x86_64'
[06/Feb/2015:10:06:33] ENGINE Error in 'start' listener <bound method CapabilitiesModel._set_capabilities of <kimchi.model.config.CapabilitiesModel object at 0x1ef4590>>
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/process/wspbus.py", line 197, in publish
    output.append(listener(*args, **kwargs))
  File "/home/christy/git/kimchi-upstream/src/kimchi/model/config.py", line 71, in _set_capabilities
    self.metadata_support = FeatureTests.has_metadata_support(conn)
  File "/home/christy/git/kimchi-upstream/src/kimchi/model/featuretests.py", line 200, in has_metadata_support
    'arch': arch})
  File "/home/christy/git/kimchi-upstream/src/kimchi/model/libvirtconnection.py", line 66, in wrapper
    ret = f(*args, **kwargs)
  File "/home/christy/git/kimchi-upstream/src/kimchi/mockmodel.py", line 164, in domainDefineXML
    return MockModel._defineXML(conn, xml)
  File "/usr/lib64/python2.7/site-packages/libvirt.py", line 2935, in defineXML
    if ret is None:raise libvirtError('virDomainDefineXML() failed', conn=self)
libvirtError: internal error: No guest options available for arch 'x86_64'

[06/Feb/2015:10:06:33] ENGINE Shutting down due to error in start listener:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/process/wspbus.py", line 235, in start
    self.publish('start')
  File "/usr/lib/python2.7/site-packages/cherrypy/process/wspbus.py", line 215, in publish
    raise exc
ChannelFailures: libvirtError("internal error: No guest options available for arch 'x86_64'",)

==========
$ cat /etc/issue
Fedora release 20 (Heisenbug)

$ uname -a
Linux christy-toshiba 3.17.8-200.fc20.x86_64 #1 SMP

$ rpm -q libvirt
libvirt-1.1.3.8-1.fc20.x86_64

**Comment by alinefm** [comment]
@clnperez Please, verify on Fedora 21 as Fedora 20 is not in our test map anymore.

**Comment by clnperez** [comment]
Yes, this is still an issue with Fedora 21.

**Comment by alinefm** [comment]
@clnperez I can not reproduce it in my Fedora 21 system.
Seems it is something specific with your system as I did a fresh Fedora 21 install.

**Comment by clnperez** [comment]
Interesting. I just installed a fresh Fedora 21 VM on my laptop and I still get the same issue.

**Comment by clnperez** [comment]
The test driver on my system gives the capabilties as:

  <guest>
    <os_type>hvm</os_type>
    <arch name='i686'>
      <wordsize>32</wordsize>
      <emulator>/usr/bin/test-hv</emulator>
      <domain type='test'>
      </domain>
    </arch>
    <features>
      <pae/>
      <nonpae/>
    </features>
  </guest>

But the regular driver gives x86_64.

**Comment by ramonmedeiros** [comment]
Can you put the output of virsh capabilities?

**Comment by clnperez** [comment]
I opened a libvirt bug with RH, and will see what they say. https://bugzilla.redhat.com/show_bug.cgi?id=1191229

**Comment by clnperez** [comment]
@ramonmedeiros here's part of it:
```<capabilities>
  <host>
    <uuid>49ff6dee-650d-4c05-9305-564424415cdb</uuid>
    <cpu>
      <arch>x86_64</arch>
      <model>SandyBridge</model>
      <vendor>Intel</vendor>
      <topology sockets='1' cores='2' threads='2'/>
...
  </host>

  <guest>
    <os_type>hvm</os_type>
    <arch name='x86_64'>
      <wordsize>64</wordsize>
      <emulator>/usr/bin/qemu-system-x86_64</emulator>
 ...
   </guest>
```

If you want the entire thing, I can paste that as well. I'm just guestting that's all you need to see.

**Comment by laggarcia** [comment]
I hit the same issue on my x86_64 machine.

**Comment by alinefm** [comment]
@clnperez I was able to reproduce it when running only test_mockmodel.
I am not sure why the error is not shown when running all the tests cases.
Anyway, I am working on a patch for it.

**Comment by alinefm** [comment]
Patch sent to Kimchi ML: [PATCH] Fix issue #591: Get the right arch for MockModel
